src/Command/RunCommand.php

```diff
@@ -668,4 +681,31 @@ private function getThreadCount(InputInterface $input): int
         // we subtract 1 here to not use all the available cores by Infection
         return max(1, CpuCoresCountProvider::provide() - 1);
     }
+
+    private function getMapSourceClassToTest(InputInterface $input): ?string
+    {
+        $inputValue = $input->getOption(self::OPTION_MAP_SOURCE_CLASS_TO_TEST);
+
+        // `false` means the option was not provided at all -> user does not care and it will be auto-detected
+        // `null` means the option was provided without any argument -> user wants to enable it
+        // any string: the argument provided, but only `'simple'` is allowed for now
+        if ($inputValue === false) {
+            return null;
+        }
+
+        if ($inputValue === null) {
+            return MapSourceClassToTestStrategy::SIMPLE;
+        }
+
+        if (is_string($inputValue) && !in_array($inputValue, [MapSourceClassToTestStrategy::SIMPLE], true)) {
```

Comment:
Do we need use in_array with just one item in haystack? simple comparison will work either :)

Reply:
In this case you could even do:

```suggestion
        if (MapSourceClassToTestStrategy::SIMPLE === $inputValue) {
```

alternatively you can skip the `is_string()` as the `in_array()` has the strict parameter type. If it was the wrong type to begin with, we are in the failure scenario in which case the extra "unnecessary" looping doesn't matter performance wise

Reply:
> Do we need use in_array with just one item in haystack? simple comparison will work either :)

I did it on purpose to show that there will be more strategies. Rewrote those to use a method that returns an array.